.circleci/config.yml

```diff
@@ -0,0 +1,515 @@
+            mkdir -p "$TEST_DATA_PATH"
+
+            TEST_SRC_DIRECTORY='src/test/java'
+            TEST_SEARCH_PATTERN='**/*Test.java'
+
+            SUREFIRE_CLASS_NAMES_PATH="${TEST_DATA_PATH}/surefire_classnames"
+
+            # generate list of tests to be processed
+            circleci tests glob "${TEST_SRC_DIRECTORY}/${TEST_SEARCH_PATTERN}" | # fancy circleci find command
+              sed -e "s#^${TEST_SRC_DIRECTORY}/\(.*\)\.java#\1#" | # Strip .java and parent directory path
+              tr "/" "." |  #Convert to a class name of form org.package.ClassName
+              grep -v '.*RouteTest' > "${SUREFIRE_CLASS_NAMES_PATH}" #Exclude RouteTest. These done from suite
+
+            # Test file name pattern does not pick this one up. Add explicitly
+            echo 'org.broadinstitute.ddp.route.IntegrationTestSuite'  >> $SUREFIRE_CLASS_NAMES_PATH
+
+            echo "Total number of test class names collected: $(wc -l $SUREFIRE_CLASS_NAMES_PATH)"
+
+            # tests split will figure out which tests to run in this node
+            THIS_NODE_TESTS_PATH=/tmp/this_node_tests
+            cat $SUREFIRE_CLASS_NAMES_PATH | \
+              circleci tests split --split-by=timings --timings-type=classname > $THIS_NODE_TESTS_PATH
+
+
+            IGNORE_CLASS_LIST_PATH="${TEST_DATA_PATH}/surefire_classnames_ignore_list"
+
+            # Filter out the ones that don't run in this node
+            cat $SUREFIRE_CLASS_NAMES_PATH | \
+              grep -xvf $THIS_NODE_TESTS_PATH > $IGNORE_CLASS_LIST_PATH
+
+            // save path for next steps
+            echo "export IGNORE_CLASS_LIST_PATH=${IGNORE_CLASS_LIST_PATH}" >> $BASH_ENV
+            source $BASH_ENV
+            echo "Total number of test classes that will be skipped in this node: $(wc -l $IGNORE_CLASS_LIST_PATH)"
+      - store_artifacts:
+          path: << parameters.circleci_path>>/tests/
+      - setup-test-dbs
+      - restore_cache: *m2_cache_keys
+      - run:
+          name: Run tests
+          command: |
+            echo "The value of IGNORE_CLASS_LIST_PATH=${IGNORE_CLASS_LIST_PATH}"
+            mvn --batch-mode -Dcheckstyle.skip test -f pom.xml \
+                    -Ditext.license=output-config/itextkey.xml \
+                    -Dddp.firecloudKeysDir=output-config/fc_keys \
+                    -Dconfig.file=output-config/testing-circleci.conf  \
+                    -Dmaven.repo.local=<< parameters.m2_repo >> \
+                    -Dsurefire.excludesFile=$IGNORE_CLASS_LIST_PATH   # Key line to do tests in parallel
+
+      - store_test_results:
+          path: target/surefire-reports
+
+  store-jar-job:
+    executor:
+      name: build-executor
+    working_directory: *pepper_apis_path
+    steps:
+      - attach_workspace:
+          at: *root_path
+      - setup-shared-env
+      - store-jar:
+          jar_base_name: DataDonationPlatform
+      - store-jar:
+          jar_base_name: Housekeeping
+
```

Comment:
Self explanatory. Notice that we reattach the workspace we saved in the compile job.